### PR TITLE
Add sqlvirtualmachine cmdlets

### PR DESCRIPTION
**Pull request by ranishan**
## Description

Cmdlet design review: Azure/azure-powershell-cmdlet-review-pr#348

Original Pull request : https://github.com/Azure/azure-powershell/pull/9985

## Checklist

- [x] I have read the [_Submitting Changes_](../CONTRIBUTING.md#submitting-changes) section of [`CONTRIBUTING.md`](../CONTRIBUTING.md)
- [x] The title of the PR is clear and informative
- [x] The appropriate `ChangeLog.md` file(s) has been updated:
    - For any service, the `ChangeLog.md` file can be found at `src/{{SERVICE}}/{{SERVICE}}/ChangeLog.md`
    - A snippet outlining the change(s) made in the PR should be written under the `## Upcoming Release` header -- no new version header should be added
- [x] The PR does not introduce [breaking changes](../documentation/breaking-changes/breaking-changes-definition.md)
- [x] If applicable, the changes made in the PR have proper test coverage
- [ ] For public API changes to cmdlets:
    - [x] a cmdlet design review was approved for the changes in [this repository](https://github.com/Azure/azure-powershell-cmdlet-review-pr) (_Microsoft internal only_)
    - [x] the markdown help files have been regenerated using the commands listed [here](../documentation/development-docs/help-generation.md#updating-all-markdown-files-in-a-module)

### CODE REVIEW

**Comment by adxsdkps** [comment]
Can one of the admins verify this patch?

**Comment by isra-fel** [comment]
@ranishan what is the relationship between this PR and #9985 ?

**Comment by ranishan** [comment]
> @ranishan what is the relationship between this PR and #9985 ?

It have created this PR as the dev for #9885 will not be able to work on this. I have fetched the code from that PR and have made changes to resolve comments given in PR #9985

**Comment by isra-fel** [comment]
> > @ranishan what is the relationship between this PR and #9985 ?
> 
> It have created this PR as the dev for #9985 will not be able to work on this. I have fetched the code from that PR and have made changes to resolve comments given in PR #9985

OK. I'll close that one then.

**Comment by ranishan** [comment]
@isra-fel @markcowl  @cormacpayne @yareyes @pratraw 
The PR is ready for review. I have resolved the test failures and comments from Mark

**Comment by ranishan** [comment]
> > > @ranishan what is the relationship between this PR and #9985 ?
> > 
> > 
> > It have created this PR as the dev for #9985 will not be able to work on this. I have fetched the code from that PR and have made changes to resolve comments given in PR #9985
> 
> OK. I'll close that one then.
> Please let us know when you're ready for review. Thanks

Ready for review. Thanks.

**Comment by ranishan** [comment]
Accidently closed. Re-opening.
Ready for review now,

File: src/SqlVirtualMachine/SqlVirtualMachine/Common/AzureSqlVirtualMachineCmdletBase.cs [hunk 1]
```diff
@@ -0,0 +1,150 @@
...
+        /// Get the ResourceId property value of the model provided.
+        /// </summary>
+        /// <param name="model"></param>
+        /// <returns></returns>
+        protected virtual string GetResourceId(M model)
+        {
+            var resource = model as IEnumerable<object>;
+            if (resource != null)
+            {
+                if (resource != null)
```
Comment (isra-fel) [inline src/SqlVirtualMachine/SqlVirtualMachine/Common/AzureSqlVirtualMachineCmdletBase.cs]: Duplicated `if` statements..

File: src/SqlVirtualMachine/SqlVirtualMachine/Az.SqlVirtualMachine.psd1 [hunk 2]
```diff
@@ -0,0 +1,141 @@
...
+
+# Modules to import as nested modules of the module specified in RootModule/ModuleToProcess
+NestedModules = @('.\Microsoft.Azure.PowerShell.Cmdlets.SqlVirtualMachine.dll')
+
+# Functions to export from this module, for best performance, do not use wildcards and do not delete the entry, use an empty array if there are no functions to export.
+FunctionsToExport = @()
+
+# Cmdlets to export from this module, for best performance, do not use wildcards and do not delete the entry, use an empty array if there are no cmdlets to export.
+CmdletsToExport = 'New-AzSqlVM',
+					'Get-AzSqlVM',
```
Comment (johnpaulkee) [inline src/SqlVirtualMachine/SqlVirtualMachine/Az.SqlVirtualMachine.psd1]: nit: spacing here.

File: src/SqlVirtualMachine/SqlVirtualMachine/Common/AzureSqlVirtualMachineCmdletBase.cs (hunk 1 above)
Comment (ranishan) [reply src/SqlVirtualMachine/SqlVirtualMachine/Common/AzureSqlVirtualMachineCmdletBase.cs]: Fixed now

File: src/SqlVirtualMachine/SqlVirtualMachine/Az.SqlVirtualMachine.psd1 (hunk 2 above)
Comment (ranishan) [reply src/SqlVirtualMachine/SqlVirtualMachine/Az.SqlVirtualMachine.psd1]: Fixed now

**Comment by ranishan** [comment]
> So I followed the comments in the original PR and all of them were fixed except the following one.
> @ranishan please fix the issue and also make sure to clean up commit history.
> See: https://github.com/Azure/azure-powershell/blob/master/documentation/development-docs/cleaning-up-commits.md
> 
> P.S I fixed an format issue in ChangeLog.md

Thank you for fixing ChangeLog.md
I tried squashing some of my commits but it leads to a lot of merge issues with code that I have not changed when I pull after rebasing. 
Is this something that is mandatory? Do you want all commits to be squashed into one?

**Comment by markcowl** [comment]
@ranishan   There are some simple ways to manage your commits.  31 is a lot of commits, and some of them seem to have duplicate names.  Here  are some simple methods to fix the commits for this PR: https://github.com/Azure/azure-powershell/blob/master/documentation/development-docs/cleaning-up-commits.md#rebasing

**Comment by ranishan** [comment]
> > So I followed the comments in the original PR and all of them were fixed except the following one.
> > @ranishan please fix the issue and also make sure to clean up commit history.
> > See: https://github.com/Azure/azure-powershell/blob/master/documentation/development-docs/cleaning-up-commits.md
> > P.S I fixed an format issue in ChangeLog.md
> 
> Thank you for fixing ChangeLog.md
> I tried squashing some of my commits but it leads to a lot of merge issues with code that I have not changed when I pull after rebasing.
> Is this something that is mandatory? Do you want all commits to be squashed into one?

Done Squashing. History is under 10 commits now.

**Comment by ranishan** [comment]
> @ranishan There are some simple ways to manage your commits. 31 is a lot of commits, and some of them seem to have duplicate names. Here are some simple methods to fix the commits for this PR: https://github.com/Azure/azure-powershell/blob/master/documentation/development-docs/cleaning-up-commits.md#rebasing

I have brought them down to 7. Thank you.

**Pull request merged**